src/state/leaderboardSlice.ts

```diff
@@ -0,0 +1,45 @@
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+import { act } from "react-dom/test-utils";
+
+export interface LeaderboardState {
+  leaders: string[]
+  scores: string[]
+  requestUpdatedLeaderboard: boolean
+}
+
+export const defaultLeaderboardState: LeaderboardState = {
+    leaders: [],
+    scores: [],
+    requestUpdatedLeaderboard: true,
+};
+
+const leaderboardStateSlice = createSlice({
+  name: `playerSlice`,
+  initialState: defaultLeaderboardState,
+  reducers: {
+    setLeaderboard: (state: LeaderboardState, action: PayloadAction<any>): void => {
+      console.log("leaderboard: ", action.payload.leaderboard);
+      action.payload.leaderboard.forEach((element : any) => {
+        const [ player, score ] = element.split(":");
+        console.log("Player and score: ", player, score);
+        state.leaders?.push(player);
```

Comment:
We don't need the ? anymore since they will never be undefined

Reply:
TRUE, thank you